lib/from_honeybee/_openapi/HoneybeeMapper.rb

```diff
@@ -0,0 +1,127 @@
+# THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+# OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *******************************************************************************
+
+require 'urbanopt/scenario'
+require 'openstudio/common_measures'
+require 'openstudio/model_articulation'
+
+require 'json'
+
+module URBANopt
+    module Scenario
+      class HoneybeeMapper < SimulationMapperBase
+
+      # class level variables
+      @@instance_lock = Mutex.new
+      @@osw = nil
+      @@geometry = nil
+
+      def initialize()
+
+        # do initialization of class variables in thread safe way
+        @@instance_lock.synchronize do
+            if @@osw.nil?
+  
+              # load the OSW for this class
+              osw_path = File.join(File.dirname(__FILE__), 'honeybee_workflow.osw')
+              File.open(osw_path, 'r') do |file|
+                @@osw = JSON.parse(file.read, symbolize_names: true)
+              end
+
+            # configures OSW with extension gem paths for measures and files, all extension gems must be
+            # required before this
+            @@osw = OpenStudio::Extension.configure_osw(@@osw)
+            end
+          end
+      end
+
+      def create_osw(scenario, features, feature_names)
+
+        if features.size != 1
+          raise "TestMapper1 currently cannot simulate more than one feature"
+        end
+        feature = features[0]
+        feature_id = feature.id
+        feature_type = feature.type
+        feature_name = feature.name
+        if feature_names.size == 1
+          feature_name = feature_names[0]
+        end
+
+        # convert to hash
+        building_hash = feature.to_hash
+        
+        if building_hash.key?(:system_type)
+          system_type = building_hash[:system_type]
+        else
+          system_type = "Inferred"
+        end
+
+        # deep clone of @@osw before we configure it
+        osw = Marshal.load(Marshal.dump(@@osw))
+
+        # now we have the feature, we can look up its properties and set arguments in the OSW
+        osw[:name] = feature_name
+        osw[:description] = feature_name
+        
+        # point to measures directory in energy-model-measure
+        osw[:measure_paths] << File.join(File.dirname(__FILE__) + "/../../measures")
+
+        # store name of input JSON file for feature
+        feature_json = feature.detailed_model_filename
+
+        #TODO: Update path
+        simulation_parameter_path = File.join(File.dirname(__FILE__) + "/../../../spec/samples/model/#{feature_json}")
```

Comment:
I don't think we want to map different simulation parameters to different features in the geoJSON. There should really only be one of these simulation parameters for all of the features. Can you remove this line?